File: python/src/renting/Item.py
from src.renting.ItemId import ItemId
from src.renting.RenterId import RenterId


class Item:
    id: ItemId
    name: str
    rented_by: RenterId | None

    def __init__(self, id: ItemId, name: str, rented_by: RenterId | None):
        self.id = id
        self.name = name
        self.rented_by = rented_by

    def is_rented(self) -> bool:
        return self.rented_by is not None

File: java-rent-nullaway/src/main/java/de/richargh/billiondollar/Item.java
package de.richargh.billiondollar;

import java.util.Optional;

public class Item {

    private final ItemId id;

    private final String name;

    private final Optional<RenterId> rentedBy;

    // Primary Constructor
    public Item(ItemId id, String name, Optional<RenterId> rentedBy) {
        this.id = id;
        this.name = name;
        this.rentedBy = rentedBy;
    }

    public ItemId id() {
        return id;
    }

    public String name() {
        return name;
    }

    public boolean isRented() {
        return !rentedBy.isEmpty();
    }
}
